src/systems/filecoin_vm/actor/actor.id

```diff
@@ -74,24 +73,18 @@ type BuiltinActorID enum {
 // to Actors in the Actor Model (programming), or Objects in Object-
 // Oriented Programming, or Ethereum Contracts in the EVM.
 //
-// Questions for id language:
-// - we should not do inheritance, we should do composition.
-//   but we should make including actor state nicer.
-//
-// ActorState represents the on-chain storage actors keep.
+// ActorState represents the on-chain storage all actors keep.
 type ActorState struct {
-    // common fields for all actors
-
+    // Identifies the code this actor executes.
     CodeID
-    // use a CID here, load it in interpreter.
-    // Alternative is to use ActorState here but tricky w/ the type system
-    State                    ActorSubstateCID
-
-    Balance                  TokenAmount
-    CallSeqNum  // FKA Nonce
-
-    // Note: may be nil if actor has no public key
-    GetSignaturePublicKey()  filcrypto.PublicKey
+    // CID of the root of optional actor-specific sub-state.
+    State       ActorSubstateCID
+    // Balance of tokens held by this actor.
+    Balance     TokenAmount
+    // Expected sequence number of the next message sent by this actor.
+    // Initially zero, incremented when an account actor originates a top-level message.
+    // Always zero for other actors.
```

Comment:
Note: also incremented for ControlActor (or whichever actor we want to use for this role); see e.g. `_applyMessageBuiltinAssert` in vm_interpreter.

Reply:
The system actor will be an account actor, so this remains true.

Reply:
Ah, didn't realize the control/system actor was intended to be an account actor. Is there a reason this has to be the case? It seems confusing (and potentially hazardous), given that it places the system actor in the same category as the current family of AccountActors that may be user-created.

Reply:
Well, it's precisely an actor with no code and with `CallSeqNum` that increments. The system right now has an invariant that only account actors can inject messages into the system, and that's exactly what we need to do.

IMO it's simpler than having a new type that behaves exactly the same way as an existing thing, and updating all our invariants about that to be "xxx _and the control actor_".

Reply:
I'm not sure about that -- for one thing, the control actor shouldn't have a public key, whereas all account actors do. Also, there are a number of places where we check (or intend to check) that an actor is an AccountActor, and in such a case, the control actor would unintentionally pass that check.

Reply:
yikes. Sorry I jumped the gun. Let's finish discussion here so we can PR against master when you have a resolution.